src/Tickets/Commerce/Ticket.php

```diff
@@ -269,6 +269,9 @@ public function get_ticket( $ticket_id ) {
 			return null;
 		}
 
+		/** @todo TribeCommerceLegacy: Remove the usage of Currency from Tribe Commerce totally, leave that behind. */
+		add_filter( 'tribe_get_option_ticket-commerce-currency-code', [ tribe( Legacy_Compat::class ), 'maybe_load_currency_code_from_tribe_commerce' ] );
+
```

Comment:
Remove the filter after the method call.

Reply:
@bordoni if we remove it after the call this fails. I chose this location (after validating that we're getting a TC ticket) because if we're at this point in the request lifecycle we're definitely at a TC screen, so the filter can be applied to all subsequent calls.